lib/reversetunnel/srv.go

```diff
@@ -747,10 +740,8 @@ func (s *server) findLocalCluster(sconn *ssh.ServerConn) (*localSite, error) {
 		return nil, trace.BadParameter("empty cluster name")
 	}
 
-	for _, ls := range s.localSites {
-		if ls.domainName == clusterName {
-			return ls, nil
-		}
+	if s.localSite.domainName == clusterName {
+		return s.localSite, nil
 	}
```

Comment:
Nit: I feel like `findLocalCluster` is now a bit confusing/misleading, both in terms of naming and the error messages it produces.  Effectively its really just checking to make sure that a connection we expect to be local based on the service it advertises is actually signed by the local CA.  I think we could improve the readability and error message(s).

If an an agent not associated with the local CA attempts to create a service conn, it currently ends up producing the error `local cluster <cluster-name> not found`, which isn't super helpful.

It would be nice if instead we could return something like `agent from cluster <cluster-name> cannot register local service <service-name>`.

Proposal:

Currently `findLocalCluster` is used like this:

```golang
cluster, err := s.findLocalCluster(sconn)
if err != nil {
    return nil, nil, trace.Wrap(err)
}
/* do stuff with 'cluster' variable */
```

Lets do something like this instead:

```golang
if err := s.requireLocalAgentForConn(sconn,connType); err != nil {
    return nil, nil, trace.Wrap(err)
}
/* use 's.localSite' directly */
```